.github/workflows/pythonpublish.yml

```diff
@@ -1,36 +1,139 @@
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+
+  github-release:
+    name: Sign the g2p distribution 📦 with Sigstore and upload them to GitHub Release
+    needs:
+      - test
+      - build
+      - build-docs
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write  # IMPORTANT: mandatory for making GitHub Releases
+      id-token: write  # IMPORTANT: mandatory for sigstore
+    steps:
+    - name: Download all the dists
+      uses: actions/download-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+    - name: Sign the dists with Sigstore
+      uses: sigstore/gh-action-sigstore-python@v1.2.3
+      with:
+        inputs: >-
+          ./dist/*.tar.gz
+          ./dist/*.whl
+    - name: Update CHANGELOG
+      id: changelog
+      uses: requarks/changelog-action@v1
+      with:
+        token: ${{ github.token }}
+        tag: ${{ github.ref_name }}
+    - name: Verbose
+      env:
+        CHANGELOG: ${{ steps.changelog.outputs.changes }}
       run: |
-        python -m pip install --upgrade pip
-        pip install build twine
-        pip install -e .
-    - name: Install documentation dependencies
+        echo REF_NAME: ${{ github.ref_name }}
+        echo CHANGELOG: $CHANGELOG
+    #- name: Create Release
+    #  uses: ncipollo/release-action@v1.12.0
+    #  with:
+    #    allowUpdates: true
+    #    name: ${{ github.ref_name }}
+    #    tag: ${{ github.ref_name }}
+    #    body: ${{ steps.changelog.outputs.changes }}
+    #    token: ${{ github.token }}
+
+  deploy-docs:
+    runs-on: ubuntu-latest
+    needs:
+      - build-docs
+      - github-release
+    steps:
+    - name: Download the latest docs
+      uses: actions/download-artifact@v4
+      with:
+        name: latest-docs
+        path: site/
+    - name: Setup doc deploy
       run: |
-        pip install -r docs/requirements.txt
-    - name: Build and publish
-      env:
-        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
-        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+        git config user.name 'github-actions[bot]'
+        git config user.email 'github-actions[bot]@users.noreply.github.com'
+    - name: Deploy docs with mike 🚀
       run: |
-        python -m build --sdist --wheel
-        twine upload dist/*
+        echo -- mike deploy --push --update-aliases latest ${{ github.ref_name }} stable
+
+  publish-to-pypi:
+    name: Publish to PyPI
+    runs-on: ubuntu-latest
+    needs: github-release
+    environment:
+      name: pypi
+      url: https://pypi.org/p/g2p
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
+    steps:
+    - name: Download the distribution packages
+      uses: actions/download-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+    - name: Publish distribution 📦 to TestPyPI
```

Comment:
secrets.TEST_PYPI_API_TOKEN isn't available on g2p. I'll create one for secrets.PYPI_API_TOKEN but I don't think I have a test pypi account for making a PAT

Reply:
I have a test pypi account if you need someone to test this...

Reply:
In fact, I can just add g2p right now as a trusted publisher!

Reply:
If you want to change it to use trusted publishing, you can publish to the test.pypi.org project "gi2pi" (the name "g2p" was already taken).

Otherwise, I have an API token but I don't have permission to add it as a secret to this repository...

Reply:
oh nice - I didn't know you could do that. thanks @dhdaines !

Reply:
I agree, Aidan, we can drop test PyPI.
dhd, that would be great if we can use trusted publishing, thanks!